dulwich/porcelain.py

```diff
@@ -1753,6 +1755,76 @@ def update_head(repo, target, detached=False, new_branch=None):
             r.refs.set_symbolic_ref(b"HEAD", to_set)
 
 
+def reset_file(repo, file_path: str , target: bytes = b'HEAD'):
+    """
+    reset the file to specific commit or branch
+    Args:
+        repo: dulwich Repo object
+        file_path: file to reset, relative to the repository path
+        target: branch or commit or b'HEAD' to reset
+    """
+    if target in branch_list(repo):
+        sha = repo.refs[b'refs/heads/' + target]
+    elif target == b'HEAD':
+        sha = repo.head()
+    else:
+        sha = target
+    tree = parse_tree(repo, treeish=sha)
+
+    def get_entry(repo, tree, file_path: str):
```

Comment:
get_entry could be replaced with tree.lookup_path(repo.object_store.__getitem__, file_path)

Reply:
I have replace `get_entry` with `tree.lookup_path(repo.object_store.__getitem__, file_path)`, 
but the test rasing error in the `dulwich.tests.test_porcelain.ResetFileTests`:
```
> python -m unittest dulwich.tests.test_porcelain.ResetFileTests
...E
======================================================================
ERROR: test_resetfile_with_dir (dulwich.tests.test_porcelain.ResetFileTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "D:\Repo\dulwich\dulwich\tests\test_porcelain.py", line 1497, in test_resetfile_with_dir
  File "D:\Repo\dulwich\dulwich\porcelain.py", line 1781, in reset_file
    file_entry = tree.lookup_path(repo.object_store.__getitem__, file_path.encode())
  File "D:\Repo\dulwich\dulwich\objects.py", line 1191, in lookup_path
    mode, sha = obj[p]
  File "D:\Repo\dulwich\dulwich\objects.py", line 1058, in __getitem__
    return self._entries[name]
KeyError: b'new_dir\\foo'
```
I think we should change `parts = path.split(b"/")` with `parts = os.path.split(path)` in the [objects.py, line 1182](https://github.com/dulwich/dulwich/blob/80bffaca1dfe59a5ba57a045f3ce7122227a7647/dulwich/objects.py#L1182)